upgrade-2.x.sh

```diff
@@ -249,6 +250,176 @@ function device_type_match {
+    #  * /mnt/sysroot/inactive: the new rootfs
+    #  * /mnt/boot: the boot partition
+    #  * /tmp/resinos-image: the FIFO file that is used to transfer the resinOS image into the container
+    #  * /mnt/data/resinhup/tmp: a location to use as docker temporary directory within the container
+    # shellcheck disable=SC2016
+    docker run \
+      --rm \
+      --name resinhup \
+      --privileged \
+      -v "${temp_sysroot}":/mnt/sysroot/inactive \
+      -v /mnt/boot:/mnt/boot \
+      -v /tmp/resinos-image.docker:/resinos-image.docker \
+      -v /mnt/data/resinhup/tmp:/mnt/data/resinhup/tmp \
+      "${update_package}" \
+      /bin/bash -c 'storage_driver=$(cat /boot/storage-driver) ; DOCKER_TMPDIR=/mnt/data/resinhup/tmp/ dockerd --log-driver=journald -s=$storage_driver --data-root='"${sysroot}"'/docker -H unix:///var/run/docker-host.sock --pidfile=/var/run/docker-host.pid --exec-root=/var/run/docker-host --bip 10.114.201.1/24 --fixed-cidr=10.114.201.128/25 --iptables=false & sleep 10; echo "Starting hostapp-update"; hostapp-update -f /resinos-image.docker' \
+    || log ERROR "Update based on hostapp-update has failed..."
+}
+
+function find_partitions {
+    # Find which partition is / and which we should write the update to
+    root_part=$(findmnt -n --raw --evaluate --output=source /)
+    log "Found root at ${root_part}..."
+    case $root_part in
+        *2)
+            # on 2.x the following device types have these kinds of results for $root_part
+            # raspberrypi: /dev/mmcblk0p2
+            # beaglebone: /dev/disk/by-partuuid/93956da0-02
+            # NUC: /dev/sda2
+            root_dev=${root_part%2}
+            update_part=${root_dev}3
+            update_part_no=3
+            update_label=resin-rootB
+            ;;
+        *3)
+            root_dev=${root_part%3}
+            update_part=${root_dev}2
+            update_part_no=2
+            update_label=resin-rootA
+            ;;
+        *rootA)
+            old_label=resin-rootA
+            update_label=resin-rootB
+            root_part_dev=$(blkid | grep "${old_label}" | awk '{print $1}' | sed 's/://')
+            update_part=${root_part_dev%2}3
+            ;;
+        *rootB)
+            old_label=resin-rootB
+            update_label=resin-rootA
+            root_part_dev=$(blkid | grep "${old_label}" | awk '{print $1}' | sed 's/://')
+            update_part=${root_part_dev%3}2
+            ;;
+        *)
+            log ERROR "Unknown root partition \"$root_part\"."
+    esac
+    if [ ! -b "$update_part" ]; then
+        log ERROR "Update partition detected as ${update_part} but it's not a block device."
+    fi
+}
+
+function finish_up() {
+    sync
+    if [ "${NOREBOOT}" == "no" ]; then
+        # Reboot into new OS
+        log "Rebooting into new OS in 5 seconds..."
+        progress 100 "ResinOS: update successful, rebooting..."
+        nohup bash -c " /bin/sleep 5 ; /sbin/reboot " > /dev/null 2>&1 &
```

Comment:
Why not just `sleep 5; reboot`?

Reply:
@petrosagg you mean instead of absolute path, or instead of using "nohup"? I take the former, but let me know if I miss something. The nohup I think is there for properly `exit 0` from the script (but it's a holdover from the original version I received, so happy to change it it could be improved further.

Reply:
got it